### ISSUE TITLE
Use `ActiveRecord::Base.primary_key` instead of hardcoding `id` in all caching strategies classes

### ISSUE THREAD

**Issue by stephenchu** [orslumen/record-cache#46]
I really think this should be the case. I am unable to use the various strategies as it stands because of this issue.

What do you think?

**Comment by orslumen** [comment]
Yes, I agree that using the `primary_key` variable is correct. So far I haven't come across this requirement as the default primary key for Rails models is `id`.

**Comment by stephenchu** [comment]
Thanks for the response. The point I was trying to make is that, by using `primary_key`, you allow greater flexibility in your library for all Rails users.

Anyways, hope to see it in some future commit 🌞
